pyiron_base/archiving/export_archive.py

```diff
@@ -0,0 +1,167 @@
+    if path_base_name == "":
+        return os.path.basename(os.path.dirname(path))
+    else:
+        return path_base_name
+
+
+def update_project(
+    project_instance, directory_to_transfer,
+    archive_directory, df
+):
+    directory_to_transfer = os.path.basename(directory_to_transfer)
+    pr_transfer = project_instance.open(os.curdir)
+    dir_name_transfer = getdir(path=directory_to_transfer)
+    dir_name_archive = getdir(path=archive_directory)
+    path_rel_lst = [
+        os.path.relpath(p, pr_transfer.project_path)
+        for p in df["project"].values
+    ]
+    return [
+        os.path.join(dir_name_archive, dir_name_transfer, p)
+        if p != "."
+        else os.path.join(dir_name_archive, dir_name_transfer)
+        for p in path_rel_lst
+    ]
+
+
+def filter_function(file_name):
+    return '.h5' in file_name
+
+
+def generate_list_of_directories(
+    df_files, directory_to_transfer,
+    archive_directory
+):
+    path_rel_lst = [
+        os.path.relpath(d, directory_to_transfer)
+        for d in df_files.dirname.unique()
+    ]
+    dir_name_transfer = getdir(path=directory_to_transfer)
+    return [
+        os.path.join(archive_directory, dir_name_transfer, p)
+        if p != "." else os.path.join(archive_directory, dir_name_transfer)
+        for p in path_rel_lst
+    ]
+
+
+def compress_dir(archive_directory):
+    arch_comp_name = archive_directory+".tar.gz"
+    tar = tarfile.open(arch_comp_name, "w:gz")
+    tar.add(archive_directory, arcname=archive_directory)
+    tar.close()
+    rmtree(archive_directory)
+
+
+def copy_files_to_archive(
+    directory_to_transfer,
+    archive_directory, compressed=True
+):
+    if directory_to_transfer[-1] != '/':
+        directory_to_transfer = os.path.basename(directory_to_transfer)
+    else:
+        directory_to_transfer = os.path.basename(directory_to_transfer[:-1])
+    # print("directory to transfer: "+directory_to_transfer)
+    pfi = PyFileIndex(
+        path=directory_to_transfer,
+        filter_function=filter_function
+    )
+    df_files = pfi.dataframe[~pfi.dataframe.is_directory]
+
+    # Create directories
+    dir_lst = generate_list_of_directories(
+        df_files=df_files,
+        directory_to_transfer=directory_to_transfer,
+        archive_directory=archive_directory
+    )
+    # print(dir_lst)
+    for d in dir_lst:
+        os.makedirs(d, exist_ok=True)
+    # Copy files
+    dir_name_transfer = getdir(path=directory_to_transfer)
+    for f in df_files.path.values:
+        copyfile(
+            f, os.path.join(
+                archive_directory, dir_name_transfer,
+                os.path.relpath(f, directory_to_transfer)
+            )
+        )
+    if compressed:
+        compress_dir(archive_directory)
+
+
+def export_database(
+    project_instance, directory_to_transfer,
+    archive_directory
+):
+    # here we first check wether the archive directory is a path
+    # or a project object
+    if static_isinstance(
```

Comment:
Please use the regular isinstance function here, we should only use the static_isinstance function to replace the hasattr part.